src/main/java/org/cbioportal/cdd/service/internal/ClinicalDataDictionaryServiceImpl.java

```diff
@@ -19,207 +19,80 @@
 import org.cbioportal.cdd.model.CancerStudy;
 import org.cbioportal.cdd.model.ClinicalAttributeMetadata;
 import org.cbioportal.cdd.service.ClinicalDataDictionaryService;
+import org.cbioportal.cdd.service.util.MSKVocabStudyUtil;
 import org.cbioportal.cdd.service.exception.CancerStudyNotFoundException;
 import org.cbioportal.cdd.service.exception.ClinicalAttributeNotFoundException;
 import org.cbioportal.cdd.service.exception.ClinicalMetadataSourceUnresponsiveException;
 import org.cbioportal.cdd.service.exception.FailedCacheRefreshException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
-/**
- * Note this class relies on the ClinicalAttributeMetadataCache class which will frequently generate a
- * new cache of clinical attributes.  Each method in this class should only get the cache once
- * and should not attempt to make any modifications to the cache.
- *
- * @author Robert Sheridan, Avery Wang, Manda Wilson
- */
 @Service
+@Qualifier("servicerouter")
 public class ClinicalDataDictionaryServiceImpl implements ClinicalDataDictionaryService {
 
     @Autowired
-    private ClinicalAttributeMetadataCache clinicalAttributesCache;
+    private MSKVocabStudyUtil mskVocabStudyUtil;
 
     @Autowired
-    private LevenshteinDistanceCache levenshteinDistanceCache;
+    @Qualifier("mskvocabulary")
+    private ClinicalDataDictionaryService cddServiceMskVocabularyImpl;
 
-    private static final Logger logger = LoggerFactory.getLogger(ClinicalDataDictionaryServiceImpl.class);
+    @Autowired
+    @Qualifier("topbraid")
```

Comment:
this should be changed to knowledgesystems (plus data member name below)

Reply:
yes, but considering we are thinking of phasing it all out I would only change it if it doesn't need to be propagated too much. Don't want a case where missed replacement causes this card to stall again for a couple days

Reply:
done